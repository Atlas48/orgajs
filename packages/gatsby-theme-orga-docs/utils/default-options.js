const path = require('path')

const defaultOptions = {
  location: ['docs'],
  imageMaxWidth: 1380,
  components: {},
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
