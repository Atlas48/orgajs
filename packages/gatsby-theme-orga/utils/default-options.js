const path = require('path')

const defaultOptions = {
  contentPath: 'content',
  filter: () => true,
  pagination: 0,
  columns: 2,
  indexPath: '/',
  imageMaxWidth: 1380,
  categoryIndexPath: category => `/${category}`,
  tagIndexPath: tag => `/:${tag}:`,
  postPath: ({ category, export_file_name }) =>
    path.resolve('/', category || '', export_file_name || ''),
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
