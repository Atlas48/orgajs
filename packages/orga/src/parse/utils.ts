import { Node } from 'unist'
import { push } from '../node'
import { Lexer } from '../tokenize'
import { Parent, Token } from '../types'

export default (lexer: Lexer) => {
  const { peek, eat, save, restore } = lexer
  const collect = (stop: (n: Token) => boolean) => (container: Parent): Parent => {
    const token = peek()
    if (!token || stop(token)) return container
    eat()
    push(container)(token)
    return collect(stop)(container)
  }

  const skip = (predicate: (token: Token) => boolean): void => {
    const token = peek()
    if (token && predicate(token)) {
      eat()
      skip(predicate)
      return
    }
    return
  }

  const tryTo = <T extends Node>(
    parse: (lexer: Lexer) => T | undefined,
    action: (node: T) => void,
  ): boolean => {
    const savePoint = save()
    const node = parse(lexer)
    if (!node) {
      restore(savePoint)
      return false
    }
    action(node)
    return true
  }

  return {
    collect,
    skip,
    tryTo,
  }
}
