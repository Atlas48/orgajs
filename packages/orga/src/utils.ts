import { Token } from './types'
import { PhrasingContent } from 'orga';

const matchOperatorsRe = /[|\\{}()[\]^$+*?.]/g

export const escape = (str: string): string => {
  return str.replace(matchOperatorsRe, '\\$&')
}

export const isPhrasingContent = (token: Token): token is PhrasingContent => {
  return token.type.startsWith('text.')
    || token.type === 'footnote.reference'
    || token.type === 'link'
}
