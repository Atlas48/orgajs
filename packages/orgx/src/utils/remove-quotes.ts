export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')

export default removeQuotes
