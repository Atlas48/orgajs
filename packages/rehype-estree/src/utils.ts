export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')
