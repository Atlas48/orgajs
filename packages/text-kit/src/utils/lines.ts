import { Point, Position } from 'unist'
import { CoreAPI } from '../core'

export default <T extends CoreAPI>(core: T) => {
  const linePosition = (
    start: number | Point,
    end?: number | Point
  ): Position | undefined => {
    let result: Position | undefined

    if (typeof start === 'number') {
      result = linePosition(core.toPoint(start))
    } else {
      const s = core.bol(start.line)
      const e = core.eol(start.line)
      if (!s || !e) return undefined
      result = {
        start: s,
        end: e,
      }
    }

    if (end && result) {
      const endLR = linePosition(end)
      if (!endLR) return undefined
      result.end = endLR.end
    }

    return result
  }
  return {
    ...core,
    linePosition,
  }
}
