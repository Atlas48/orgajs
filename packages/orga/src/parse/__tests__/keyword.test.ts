import { parse } from './utils'

it('can handle TODO keywords', () => {
  expect(
    parse(
      `
#+TODO: TODO NEXT | DONE

* NEXT Some Headline
`.trim()
    )
  ).toMatchSnapshot()
})

it('can handle multiple TODO keywords', () => {
  expect(
    parse(
      `
#+TODO: TODO NEXT | DONE
#+TODO: DRAFT PUBLISHED
#+TODO: BUG FEATURE | DONE

* DRAFT Some Headline
`.trim()
    )
  ).toMatchSnapshot()
})
