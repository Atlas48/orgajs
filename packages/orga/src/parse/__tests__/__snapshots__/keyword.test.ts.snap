// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can handle TODO keywords 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "actionable": true,
          "children": Array [
            Object {
              "level": 1,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 2,
                  "line": 3,
                  "offset": 27,
                },
                "start": Object {
                  "column": 1,
                  "line": 3,
                  "offset": 26,
                },
              },
              "type": "stars",
            },
            Object {
              "actionable": true,
              "keyword": "NEXT",
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 7,
                  "line": 3,
                  "offset": 32,
                },
                "start": Object {
                  "column": 3,
                  "line": 3,
                  "offset": 28,
                },
              },
              "type": "todo",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 21,
                  "line": 3,
                  "offset": 46,
                },
                "start": Object {
                  "column": 8,
                  "line": 3,
                  "offset": 33,
                },
              },
              "type": "text.plain",
              "value": "Some Headline",
            },
          ],
          "content": "Some Headline",
          "keyword": "NEXT",
          "level": 1,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 3,
              "offset": 46,
            },
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 26,
            },
          },
          "type": "headline",
        },
      ],
      "level": 1,
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 21,
          "line": 3,
          "offset": 46,
        },
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 26,
        },
      },
      "properties": Object {},
      "type": "section",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 21,
      "line": 3,
      "offset": 46,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle multiple TODO keywords 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "actionable": true,
          "children": Array [
            Object {
              "level": 1,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 2,
                  "line": 5,
                  "offset": 78,
                },
                "start": Object {
                  "column": 1,
                  "line": 5,
                  "offset": 77,
                },
              },
              "type": "stars",
            },
            Object {
              "actionable": true,
              "keyword": "DRAFT",
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 8,
                  "line": 5,
                  "offset": 84,
                },
                "start": Object {
                  "column": 3,
                  "line": 5,
                  "offset": 79,
                },
              },
              "type": "todo",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 22,
                  "line": 5,
                  "offset": 98,
                },
                "start": Object {
                  "column": 9,
                  "line": 5,
                  "offset": 85,
                },
              },
              "type": "text.plain",
              "value": "Some Headline",
            },
          ],
          "content": "Some Headline",
          "keyword": "DRAFT",
          "level": 1,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 22,
              "line": 5,
              "offset": 98,
            },
            "start": Object {
              "column": 1,
              "line": 5,
              "offset": 77,
            },
          },
          "type": "headline",
        },
      ],
      "level": 1,
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 22,
          "line": 5,
          "offset": 98,
        },
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 77,
        },
      },
      "properties": Object {},
      "type": "section",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 22,
      "line": 5,
      "offset": 98,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;
